frontend/src/components/QueryBuilder/QueryBuilder.tsx

```diff
@@ -1427,315 +1353,60 @@ function QueryBuilder<T extends SearchParamsInput | ErrorSearchParamsInput>(
 	const getDefaultOperator = (field: SelectOption | undefined) =>
 		((field && getCustomFieldOptions(field)?.operators) ?? OPERATORS)[0]
 
-	const parseInner = useCallback(
```

Comment:
moved a lot of the logic for creating the `backendSearchQuery` to `SearchState.ts`